ui/store/controller-actions/gator-permissions-controller.ts

```diff
@@ -1,10 +1,69 @@
 import { GatorPermissionsMap } from '@metamask/gator-permissions-controller';
+import { Hex, Json } from '@metamask/utils';
+import {
+  encodeDisabledDelegationsCheck,
+  decodeDisabledDelegationsResult,
+} from '../../../shared/lib/delegation/delegation';
 import { submitRequestToBackground } from '../background-connection';
 
-export const fetchAndUpdateGatorPermissions =
-  async (): Promise<GatorPermissionsMap> => {
-    return await submitRequestToBackground(
-      'fetchAndUpdateGatorPermissions',
-      [],
-    );
-  };
+export const fetchAndUpdateGatorPermissions = async (
+  params?: Json,
```

Comment:
Maybe define params type? Probably define in core and import here so its not just Json?